From 74e7822480cdfc1b550f92d7ac7b9f58886fe2c6 Mon Sep 17 00:00:00 2001
From: Wulsic <wulsic@hotmail.com>
Date: Tue, 20 Jan 2015 15:58:51 +0100
Subject: [PATCH 1/1] Make microphone to work This patch which comes from
 Dhinesh77 fixes the issue that the microphone is'nt working in the custom
 AOSP based roms.

Change-Id: Id5c3ffdbbab937c3875b460c80c4dfac27ed5c99
---
 include/hardware/audio.h | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/include/hardware/audio.h b/include/hardware/audio.h
index 0f16484..db8e508 100644
--- a/include/hardware/audio.h
+++ b/include/hardware/audio.h
@@ -683,6 +683,20 @@ struct audio_hw_device {
     void (*close_output_stream)(struct audio_hw_device *dev,
                                 struct audio_stream_out* stream_out);
 
+#ifdef HAWAII_HWC
+ /** This method creates and opens the audio hardware output
+ * for broadcast stream */
+ int (*open_broadcast_stream)(struct audio_hw_device *dev, uint32_t devices,
+ int format, uint32_t channels,
+ uint32_t sample_rate,
+ uint32_t audio_source,
+ struct audio_broadcast_stream **out);
+
+ void (*close_broadcast_stream)(struct audio_hw_device *dev,
+ struct audio_broadcast_stream *out);
+#endif
+
+
     /** This method creates and opens the audio hardware input stream */
 #ifndef ICS_AUDIO_BLOB
     int (*open_input_stream)(struct audio_hw_device *dev,
-- 
1.9.1

